src/components/form/components/select.js

```diff
@@ -57,13 +57,21 @@ Select.propTypes = {
   disabled: PropTypes.bool,
   multiple: PropTypes.bool,
   loading: PropTypes.bool,
-  value: PropTypes.oneOfType([
-    PropTypes.string,
-    PropTypes.number,
-    PropTypes.arrayOf(
-      PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    ),
-  ]),
+  value: function(props, propName, componentName) {
+    if (props['multiple']) {
+      PropTypes.checkPropTypes(
+        { [propName]: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])) },
+        props,
+        propName,
+        componentName);
+    } else {
+      PropTypes.checkPropTypes(
+        { [propName]: PropTypes.oneOfType([PropTypes.string, PropTypes.number]) },
+        props,
+        propName,
+        componentName);
+    }
+  },
```

Comment:
is there any reason why you have decided to use a custom prop type checking function instead of:
```
PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.number,
  PropTypes.arrayOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  )
])
```
?

Reply:
Hello @MrCreeper1008, thank you for reviewing.

Using a function allowed me to remove one flaw in the initial proposition:
The prop type check is actually dependent of the `multiple` value. As you can see, if `multiple` is `true` we really want an array of strings/numbers. But in case, `multiple` is `false`, providing such array is a mistake which the initial check is not capable to catch.
I feel this function enables a more complete fix as it removes initial proposition flaw.

Also, this is the reason why I have added tests that check different combination of `multiple` and `value`. For some test cases, I actually check it throws errors in case of misconfiguration.

Reply:
you are correct, in this case asserting that no console errors will be thrown is fine.